### PR TITLE
msg: Frontend/Backend Feature

### PR DESCRIPTION
**Pull request by DanielBar-On**

### CODE REVIEW

File: src/msg/Messenger.cc [hunk 1]
```diff
@@ -14,9 +14,10 @@
 
 Messenger *Messenger::create_client_messenger(CephContext *cct, string lname)
 {
+	std::string public_msgr_type = cct->_conf->ms_public_type.empty() ? cct->_conf->ms_type : cct->_conf->ms_public_type;
```
Comment (liewegas) [inline src/msg/Messenger.cc]: whitespace

File: src/msg/async/Event.cc [hunk 2]
```diff
@@ -99,7 +99,7 @@ ostream& EventCenter::_event_prefix(std::ostream *_dout)
                 << " time_id=" << time_event_next_id << ").";
 }
 
-int EventCenter::init(int n, unsigned i, std::string &t)
+int EventCenter::init(int n, unsigned i,const std::string &t)
```
Comment (liewegas) [inline src/msg/async/Event.cc]: missing space

File: src/msg/async/Event.h [hunk 3]
```diff
@@ -191,7 +191,7 @@ class EventCenter {
   ~EventCenter();
   ostream& _event_prefix(std::ostream *_dout);
 
-  int init(int nevent, unsigned idx, std::string &t);
+  int init(int nevent, unsigned idx, const std::string &t = "");
```
Comment (liewegas) [inline src/msg/async/Event.h]: please avoid default arguments unless they are really necessary

**Comment by liewegas** [comment]
@yuyuyu101 ?

**Comment by yuyuyu101** [comment]
I will extract the fixes into original pull request

**Comment by liewegas** [comment]
ok cool. then close this one and mark yours needs-qa?

**Comment by yuyuyu101** [comment]
sure, I will complete my pr

**Pull request closed without merging**